_ml-commons-plugin/api.md

```diff
@@ -0,0 +1,968 @@
+---
+layout: default
+title: API
+has_children: false
+nav_order: 99
+---
+
+# ML Commons API 
+
+---
+
+#### Table of contents
+- TOC
+{:toc}
+
+
+---
+
+The Machine Learning (ML) commons API lets you create, train, and store machine learning algorithms both synchronously and asynchronously.  
+
+In order to train tasks through the API, three inputs are required. 
+
+- Algorithm name: Usually `FunctionaName`. This determines what algorithm the ML Engine runs.
+- Model hyper parameters: Adjust these parameters to make the model train better.  You can also implement `MLAgoParamas` to build custom parameters for each model.
```

Comment:
For REST API user, they should not build/implement new `MLAgoParamas`. They can only tune parameters what we supported, for example tune `centroids` of kmeans parameter from "2" to "3" to better cluster the data. 

For developers, if they need to add new algorithms, they need to implement `MLAgoParamas` to build custom hyper parameters for their algorithms.

Reply:
@ylwu-amzn: I'll remove the MLAgoPararms and address it in a different page. Do we have a happy path for adding algorithms? Similar to the path we have for developers to add functions?

Reply:
https://github.com/opensearch-project/ml-commons/blob/main/docs/how-to-add-new-function.md, refer to this doc to add new ML algorithm, check example1